---
title: "Fix GFF format"
author: "Ariana S Huffmyer"
date: "2022"
output: html_document
editor_options: 
  chunk_output_type: console
---
This script adds transcript and gene id into GFF file for alignment.  

Here, I'll be adding transcript_id= and gene_id= to 'gene' column because we needs that label to map our RNAseq data  

Load libraries and data. 
```{r}
#Load libraries
library(tidyverse)
library(R.utils)
```

Load  gene gff file

```{r}
gff <- read.csv(file="data/Tagseq_zymo_genohub/Pver_genome_assembly_v1.0.gff3", header=FALSE, sep="\t", skip=1) 
#gff <- na.omit(gff)

#gff.DB <- read.csv(file="data/Tagseq_zymo_genohub/Pver_genome_assembly_v1.0_modified.gff3", header=FALSE, sep="\t", skip=1) 
#gff.DB <- na.omit(gff.DB)
```

Rename columns 
```{r}
colnames(gff) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
head(gff)

unique(gff$scaffold)
#shows that #PROT and # PASA_UPDATE # ORIGINAL: lines have been included but only have information in column 1 scaffold
```

Remove extra #PROT and # PASA_UPDATE and # ORIGINAL:
check the number of genes against the number in the genome
```{r}
#genes in the Pverrucosa genome 27,439

# remove any rows with NA
gff <- na.omit(gff)

# check the number of genes against the number in the genome
# Pverrucosa should return 27439
nrow(gff %>%filter(id=="gene"))

#genes <- gff %>%filter(id=="mRNA")
#genes$gene <- gsub("Pver_gene_split_gene_g", "Pver_split_gene_g", genes$gene) #remove extra "_gene_" from name

```

Remove cds. and duplication of _gene in the names
```{r}
#gff$ID <- sub(";.*", "", gff$gene) #move everything before the first ";" to a new column
#gff$ID <- gsub("cds.", "", gff$gene) #remove "cds." from the new column
gff$ID <- gsub("Pver_gene_g", "Pver_g", gff$gene) #remove "_gene" from the new column
gff$ID <- gsub("Pver_gene_split_gene_g", "Pver_split_gene_g", gff$ID) #remove extra "_gene_" from name
gff$ID <- gsub("Pver_gene_novel_model_", "Pver_novel_model_", gff$ID) #remove extra "_gene_" from name

head(gff)
```

check if gene names are all found in the Annotation file
```{r}
genes <- gff %>%filter(id=="gene") #filter to only the genes
genes$Query <- sub("^.*?\\;", "", genes$ID) # keep the Parent and Name information
genes$Query <- sub("^.*?\\;", "", genes$Query) # keep the Name information
genes$Query <- gsub("Name=", "", genes$Query) #remove "Name=" from the column
genes$check <- genes$Query

#Read in annotation file to check gene names are the same in the gff and the "Query" column of the annotation
Annot <- readxl::read_xlsx("data/Tagseq_zymo_genohub/FileS2_Pver_gene_annot_May28.xlsx", skip=4) 

#see if the number of rows is the same
nrow(Annot)==nrow(genes)

#how many unique gene names do they each have
length(unique(Annot$Query))
length(unique(genes$Query))


length(unique(Annot$Query))- length(unique(genes$Query))

diffs <-left_join(Annot, genes, by="Query")


```

Save file. Gzip and then upload this to the server for use in bioinformatic steps.  
```{r}
write.table(gff, file="data/Tagseq_zymo_genohub/Pver_genome_assembly_v1.0_fixed.gff3", sep="\t", col.names = FALSE, row.names=FALSE, quote=FALSE)

#gzip the file 
#gzip("data/Tagseq_zymo_genohub/Pver_genome_assembly_v1.0_fixed.gff3")
```


