---
title: "GeneExpression"
author: "HM Putnam"
date: "4/24/2024"
output: html_document
---

# Install and Load Packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

library("tidyverse")
library("DESeq2")
library("genefilter")
library("RColorBrewer")
library("pheatmap")
library("viridis")
library("ggplot2")
library("goseq")
library("strex")
library("ggpubr")
library("ggplotify")
library("EnhancedVolcano")
library("enrichplot")
library("rrvgo")
library("ggtree")
library("tidytree")
library("clusterProfiler")
library("org.Hs.eg.db")
library("GOSemSim")
library("patchwork")

```

#Read in data and metadata
```{r}
#read in metadata
sample_ColData <- read.csv("data/Tagseq_zymo_genohub/Pverr_larvae_Seq_Sample_Info.csv", header=TRUE, sep=",")

#generate a grouping variable
sample_ColData$group <- paste0(sample_ColData$Stage,"_",sample_ColData$Temperature)

#Read in data
TranscriptCountData <- as.data.frame(read.csv(file= "data/Tagseq_zymo_genohub/Pverr_larvae_gene_count_matrix.csv", sep=",", header = TRUE))

#view data
head(TranscriptCountData)

#assign the first row of gene names as rownames
rownames(TranscriptCountData) <- TranscriptCountData[,1]

#remove gene names so only numeric counts data are in dataset
TranscriptCountData <- TranscriptCountData[,-1]

#view data
head(TranscriptCountData)
```

#Functional Annotation
```{r}
# Read in Functional Annotation
Annot <- readxl::read_xlsx("data/Tagseq_zymo_genohub/FileS2_Pver_gene_annot_May28.xlsx", skip=4) #read in data file

```

#filtering values for PoverA
```{r}

#filter data to embryo and larval stages at 28°
TranscriptCountData <- TranscriptCountData[,c(1:5,7,9,11)]

#filter sample metadata to embryo and larval stages at 28°
sample_ColData <- sample_ColData[c(1:5,7,9,11),]

#set filter values for PoverA, P percent of the samples have counts over A
filt <- filterfun(pOverA(0.5,10))

#create filter for the counts data
tfil <- genefilter(TranscriptCountData, filt)

#identify transcripts to keep by count filter
keep <- TranscriptCountData[tfil,]

#identify transcript list
gn.keep <- rownames(keep)

#data filtered in PoverA, P percent of the samples have counts over A
filt.count.mat <- as.matrix(TranscriptCountData[which(rownames(TranscriptCountData) %in% gn.keep),])

#view data
head(filt.count.mat)
```

#check count matrix sample name order (column order) and sample metadata order matches
```{r}
#assign tube ids as rownames in metadata
rownames(sample_ColData) <- sample_ColData$Tube_ID

#view sample names in data
colnames(filt.count.mat) 

#view sample names in metadata
rownames(sample_ColData)

# Check all sample IDs in sample_ColData are also in TranscriptCountData and match their orders
all(rownames(sample_ColData) %in% colnames(filt.count.mat))  #Should return TRUE
all(rownames(sample_ColData) == colnames(filt.count.mat))    #Should return TRUE

#identify the treatment column levels
sample_ColData$group <- factor(sample_ColData$group)

#view levels
levels(sample_ColData$group)

```

# Contruct expression object
```{r}
#### Construct DESeq dataset from matrix
data <- DESeqDataSetFromMatrix(countData = filt.count.mat, colData = sample_ColData, design = ~ group) #create a DESeqDataSet object

#apply a regularized log transformation to minimize effects of small counts and normalize wrt library size
rld <- rlog(data, blind=FALSE)

#view data
head(assay(rld), 3)

#calculate distance matix
sampleDists <- dist(t(assay(rld)))

#distance matrix
sampleDistMatrix <- as.matrix(sampleDists)

#assign col names
colnames(sampleDistMatrix) <- sample_ColData$group

#assign colors
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)

#### Plotting heatmap for Expression Visualization of all genes in each sample
pheatmap(sampleDistMatrix, #plot matrix of expression similarity
         clustering_distance_rows=sampleDists, #cluster rows
         clustering_distance_cols=sampleDists, #cluster columns
         col=colors) #set colors

#### Plotting PCA from Expression visualization
plotPCA(rld, intgroup = c("group")) #plot PCA of samples with all data

```



# Differential Gene Expression Analysis
```{r}
#DESeq2 Test: test of the factor 

#run differential expression test by group using the wald test
DEG.int <- DESeq(data)

#save DE results
DEG.int.res <- results(DEG.int)

# Volcano Plot (significance as a function of fold change)
par(mar=c(5,5,5,5), cex=1.0, cex.main=1.4, cex.axis=1.4, cex.lab=1.4)
topT <- as.data.frame(DEG.int.res)
#Adjusted P values (FDR Q values)
with(topT, plot(log2FoldChange, padj, pch=20, main="Volcano plot", cex=1.0, xlab=bquote(~Log[2]~fold~change), ylab=bquote(padj)))
abline(h=0.05, col="black", lty=3, lwd=1.0)
#Color the significant points with log2 fold change >2 red ()
with(subset(topT, padj<0.05), points(log2FoldChange, padj, pch=20, col="red", cex=0.5))


  EnhancedVolcano(DEG.int.res,
    lab = rownames(DEG.int.res),
    x = 'log2FoldChange',
    y = 'pvalue')

#view DE results
resultsNames(DEG.int)

# save log fold change information
log2_changes <- DEG.int.res[,c(2,6)]

#identify the number of significant p values with 5%FDR (padj<0.05)
sig.num <- sum(DEG.int.res$padj <0.05, na.rm=T)

#identify signficant pvalues with 5%FDR
sig <- subset(DEG.int.res, padj<0.05,)

#subset list of sig transcripts from original count data
sig.list <- data[which(rownames(data) %in% rownames(sig)),]

#save DEG list
write.csv(counts(sig.list), file="output/DEG_05FDR.Stage.28.csv")

#save DEG df
res_df <- as.data.frame(DEG.int.res)
res_df <- res_df %>% 
  subset(padj<0.05)

#subset upregulated in Embryos
up.EMB.sig <- res_df %>% 
  subset(padj<0.05) %>%
  subset(log2FoldChange<0) 

#Sanity check on expression direction 
plotCounts(DEG.int, gene=rownames(up.EMB.sig)[1], intgroup="group") 

#subset list of sig transcripts from original count data
up.EMB.sig.list <- data[which(rownames(data) %in% rownames(up.EMB.sig)),]
up.EMB.sig.list <- as.data.frame(counts(up.EMB.sig.list))

#add annotation data
up.EMB.sig.list$Gene <- rownames(up.EMB.sig.list)
up.EMB.sig.list.annot <- left_join(up.EMB.sig.list, Annot,  by="Gene")

#subset upregulated at Larval stage
up.LAR.sig <- res_df %>% 
  subset(padj<0.05) %>%
  subset(log2FoldChange>0) 

#Sanity check on expression direction. 
plotCounts(DEG.int, gene=rownames(up.LAR.sig)[1], intgroup="group") 

#subset list of sig transcripts from original count data
up.LAR.sig.list <- data[which(rownames(data) %in% rownames(up.LAR.sig)),]
up.LAR.sig.list <- as.data.frame(counts(up.LAR.sig.list))

#add annotation data
up.LAR.sig.list$Gene <- rownames(up.LAR.sig.list)
up.LAR.sig.list.annot <- left_join(up.LAR.sig.list, Annot,  by="Gene")


```

#Plotting Differentially Expressed Genes (DEGs) PCA
```{r}
#### Plotting DEG PCA

# set group colors
cols <- c("Embryo_28" = "coral","Larvae_28" = "lightblue")

# rlog transform data
rsig <- rlog(sig.list, blind=FALSE)

#Plot PCA of all samples for DEG only
PCA.plot.auto <- plotPCA(rsig, intgroup = c("group"))
#view plot
PCA.plot.auto 

#extract plotting data
PC.info <-PCA.plot.auto$data 

#plot PCA results
#pdf(file="figures/Pverr_larvae_stage_PCA.DEG.pdf")
PCA.plot <- PC.info %>%
ggplot(aes(x=PC1, y=PC2,color=group), group=group)+
  geom_point(size=3)+
  scale_color_manual(values=cols)+
  xlab(paste0(PCA.plot.auto$labels$x[1]))+
  ylab(paste0(PCA.plot.auto$labels$y[1]))+
  theme_bw()+
  theme(legend.position = c(0.8, 0.4),
         text = element_text(size=15))
PCA.plot 
#dev.off()

```
#### Plotting DEG heatmap
```{r}
#### Plotting DEG heatmap

# set group colors
ann_colors = list(
  group = c("Embryo_28" = "coral","Larvae_28" = "lightblue"))

#make an expression object
mat <- assay(rsig)#

#make dataframe
df <- data.frame(colData(rsig)[c("group")])

#plot heatmap by stage
#pdf("output/Pverr_larvae_Stage_DEG_heatmap_relative.pdf")
Hmap <- pheatmap(mat, color=inferno(10), annotation_col = df, annotation_colors = ann_colors, clustering_method = "average", scale="row",  clustering_distance_rows="euclidean", show_rownames =F, fontsize_row=8, cluster_cols=T, show_colnames =F, gaps_col = c(5)) #plot heatmap of all DEG by group
Hmap 
#dev.off()

```
#Plotting PCA and Heatmap together
```{r}
#arrange plots
DEplots <- ggarrange(PCA.plot, as.grob(Hmap), ncol=2, labels = c("A)","B)"))
#save file
ggsave("figures/Pverr_larvae_DEplots_Stage.pdf", DEplots, width = 16, height=8)
ggsave("figures/Pverr_larvae_DEplots_Stage.png", DEplots, width = 16, height=8)

```


#Adding Annotation to DEGs
```{r}
#identify DEGs
DEGs <- res_df
#set gene name as rowname
DEGs$Gene <- rownames(DEGs)
#join counts and annotation
Output <- left_join(DEGs,Annot,  by="Gene")
#save file of annotated DEGs
write.csv(Output,"output/Annotated_DEGS.csv")

#separate short hit name from long name
Output <- Output %>% separate(Hit_description, into = c("Hit_description1", "Hit_description2"),
          sep = "\\(|\\)")

#make a label name of hit name and gene ID
Output$label <- paste0(Output$Hit_description1, " (", Output$Gene, ")")

#remove the gene ID Pver_gene_ to leave the gene number
Output$label <- gsub("Pver_gene_", "", Output$label)

#set the order of factors for labels
Output$label <- factor(Output$label, levels = Output$label[order(Output$log2FoldChange)])

#make a data frame for annotation
data2.labels <- data.frame(
  x = c(14, 11),
  y = c(6, -6),
  padj=c(0,0),
  text.lab = c("Higher in Embryos", "Higher in Larvae")
  )

#plot fold change higher in larvae and higher in embryos
#pdf("figures/DEG_LFC_Larvae_Stage.pdf", height = 12, width = 24)
DEG_LFC <- Output %>% arrange(log2FoldChange) %>%
  ggplot(aes(x=label, y=log2FoldChange, colour=padj)) +
  #ylim(-10,10)+
  geom_point(size=2)+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=label, xend=label, y=0, yend=log2FoldChange),linewidth=1) +
  coord_flip()+
  theme_classic() +
  theme(legend.position = c(0.8, 0.1),
        axis.title.y = element_blank(),
        axis.title.x = element_text(size = 14, face = "bold"),
        plot.title = element_text(size = 40, face = "bold"))+
  ggtitle('A')
#DEG_LFC
dev.off()

```
# GO ENRICHMENT
# calculate Gene Length
```{r}
#read in the genome gff file
calc.lengths <- read.table("data/Tagseq_zymo_genohub/Pver_genome_assembly_v1.0_fixed.gff3", header=F)

#filter to the mRNA
calc.lengths <- calc.lengths %>% filter(V3=="mRNA")

#calculate the length from the stop and start values
calc.lengths$length <- calc.lengths$V5 - calc.lengths$V4

#trim the IDs to the gene name only
calc.lengths$V9 <- str_after_nth(calc.lengths$V9, "\\;", 1)
calc.lengths$V9 <- gsub("geneID=", "", calc.lengths$V9)

#subset the dataframe to gene name and gene length
calc.lengths <- calc.lengths[,c(9:10)]

#rename the column
colnames(calc.lengths)[1] <- "Gene"
```

#GO enrichment of DEGs that are HIGHER IN EXPRESSION in Larvae
```{r}
#set the all transcripts list
ALL<-row.names(data)

#set the enrichment test list with the expressed filtered genes
DEG <- as.character(up.LAR.sig.list$Gene) 

#set ID names
IDs <- row.names(data) 

#set length information
LENGTH <-calc.lengths 

#load GO terms for all genes in the genome
GO.terms <-Annot[,c(2,13)]

#split into multiple GO ids
splitted <- strsplit(as.character(GO.terms$Gene_ontology_IDs), ";") 

#list all GOs with each assigned gene
GO.terms <- data.frame(v1 = rep.int(GO.terms$Gene, sapply(splitted, length)), v2 = unlist(splitted)) 

#remove empty spaces before GO
GO.terms$v2 <- gsub(" GO:", "GO:",GO.terms$v2) 

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DEG.vector <-c(t(DEG))
ID.vector <- LENGTH$Gene
LENGTH.vector <-LENGTH$length

#Construct new vector with 1 for DEG and 0 for others
gene.vector <-  as.integer(ID.vector%in%DEG.vector) 

#set names
names(gene.vector)=ID.vector 

#weight vector by length of gene
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#identify enriched GO terms
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at chosen cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.01]

#format as dataframe
enriched.GO.05<-data.frame(enriched.GO.05.a)

#rename column
colnames(enriched.GO.05) <- c("category")

#include enrichment information for each category (GO Term)
enriched.GO.05 <- left_join(enriched.GO.05, GO.wall, by="category")


#subset the Molecular Functions
LAR.MF.INT <- subset(enriched.GO.05, ontology=="MF")
#order by the number in category
LAR.MF.INT <- LAR.MF.INT[order(LAR.MF.INT$numDEInCat),]
#subset the Cellular Components
LAR.CC.INT <- subset(enriched.GO.05, ontology=="CC")
#order by the number in category
LAR.CC.INT <- LAR.CC.INT[order(LAR.CC.INT$numDEInCat),]
#subset the Biological Processes
LAR.BP.INT <- subset(enriched.GO.05, ontology=="BP")
#order by the number in category
LAR.BP.INT <- LAR.BP.INT[order(LAR.BP.INT$numDEInCat),]
write.csv(enriched.GO.05 , file = "output/DEGS_Up_in_Larvae_vs_Emb_Enriched_GOTerms.csv")

```

#Plotting BP enriched in DEGS up.EMB.sig.list
```{r}
# similarity of GO terms in the DEG list for BP
simMatrix <- calculateSimMatrix(LAR.BP.INT$category,
                                orgdb="org.Hs.eg.db", #human database
                                ont=c("BP"),
                                method="Rel")

#scoring sig enriched terms by pvalue
scores <- setNames(-log(LAR.BP.INT$over_represented_pvalue), LAR.BP.INT$category)

#reducing sig enriched terms to higher order terms and grouping them together by a threshold value
# and weighting them by scores of pvalue
#higher thresholds result in fewer groups
reducedTerms.BP.LAR <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.9,
                                orgdb="org.Hs.eg.db")
colnames(reducedTerms.BP.LAR)[1] <- "category"

pdf("figures/BP_DEGS_Up_in_Larvae_vs_Emb_Enriched_GOTerms_semsim.pdf", width = 18, height = 8)
treemapPlot(reducedTerms.BP.LAR, force.print.labels=T)
dev.off()
```
#Plotting MF enriched in DEGS up.EMB.sig.list
```{r}
# similarity of GO terms in the DEG list for BP
simMatrix <- calculateSimMatrix(LAR.MF.INT$category,
                                orgdb="org.Hs.eg.db", #human database
                                ont=c("MF"),
                                method="Rel")

#scoring sig enriched terms by pvalue
scores <- setNames(-log(LAR.MF.INT$over_represented_pvalue), LAR.MF.INT$category)

#reducing sig enriched terms to higher order terms and grouping them together by a threshold value
# and weighting them by scores of pvalue
#higher thresholds result in fewer groups
reducedTerms.MF.LAR <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.9,
                                orgdb="org.Hs.eg.db")
colnames(reducedTerms.MF.LAR)[1] <- "category"

pdf("figures/MF_DEGS_Up_in_Larvae_vs_Emb_Enriched_GOTerms_semsim.pdf", width = 18, height = 8)
treemapPlot(reducedTerms.MF.LAR, force.print.labels=T)
dev.off()
```