---
title: "GeneExpression"
author: "HM Putnam"
date: "4/24/2024"
output: html_document
---

Install and Load Packages
```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)

#library("tidyr")
library("tidyverse")
library("DESeq2")
library("genefilter")
library("RColorBrewer")
library("pheatmap")
library("viridis")
library("ggplot2")
library("goseq")
library("strex")
library("ggpubr")
library("ggplotify")
library("EnhancedVolcano")
library("enrichplot")
library("rrvgo")
library("ggtree")
library("tidytree")
library(clusterProfiler)
library(org.Hs.eg.db)
library(GOSemSim)
library(DOSE)

```

#Read in data and metadata
```{r}
#read in metadata
sample_ColData <- read.csv("data/Tagseq_zymo_genohub/Pverr_larvae_Seq_Sample_Info.csv", header=TRUE, sep=",")
print(sample_ColData)
sample_ColData$group <- paste0(sample_ColData$Stage,"_",sample_ColData$Temperature)

#Read in data
TranscriptCountData <- as.data.frame(read.csv(file= "data/Tagseq_zymo_genohub/Pverr_larvae_gene_count_matrix.csv", sep=",", header = TRUE))
head(TranscriptCountData)

rownames(TranscriptCountData) <- TranscriptCountData[,1]
TranscriptCountData <- TranscriptCountData[,-1]
head(TranscriptCountData)
```

#Functional Annotation

```{r}
# Functional Annotation
Annot <- readxl::read_xlsx("data/Tagseq_zymo_genohub/FileS2_Pver_gene_annot_May28.xlsx", skip=4) #read in data file

```

#filtering values for PoverA
```{r}

#filter to larval stage
TranscriptCountData <- TranscriptCountData[,c(5:12)]
sample_ColData <- sample_ColData[c(5:12),]

#set filter values for PoverA, P percent of the samples have counts over A
filt <- filterfun(pOverA(0.5,10))

#create filter for the counts data
tfil <- genefilter(TranscriptCountData, filt)

#identify transcripts to keep by count filter
keep <- TranscriptCountData[tfil,]

#identify transcript list
gn.keep <- rownames(keep)

#data filtered in PoverA, P percent of the samples have counts over A
filt.count.mat <- as.matrix(TranscriptCountData[which(rownames(TranscriptCountData) %in% gn.keep),])
head(filt.count.mat)
```

#check count matrix sample name order (column order) and sample metadata order matches
```{r}
#assign tube ids as rownames in metadata
rownames(sample_ColData) <- sample_ColData$Tube_ID

rownames(sample_ColData)
colnames(filt.count.mat) 

# Check all sample IDs in sample_ColData are also in TranscriptCountData and match their orders
all(rownames(sample_ColData) %in% colnames(filt.count.mat))  #Should return TRUE
# returns TRUE
all(rownames(sample_ColData) == colnames(filt.count.mat))    # should return TRUE
#returns TRUE

#identify the treatment column levels
sample_ColData$group <- factor(sample_ColData$group)
levels(sample_ColData$group)

```


```{r}
#### Construct DESeq dataset from matrix
data <- DESeqDataSetFromMatrix(countData = filt.count.mat, colData = sample_ColData, design = ~ group) #create a DESeqDataSet object


#apply a regularized log transformation to minimize effects of small counts and normalize wrt library size
rld <- rlog(data, blind=FALSE)
head(assay(rld), 3) #view data

#calculate distance matix
sampleDists <- dist(t(assay(rld)))

#distance matrix
sampleDistMatrix <- as.matrix(sampleDists)

#assign col names
colnames(sampleDistMatrix) <- sample_ColData$group

#assign colors
colors <- colorRampPalette( rev(brewer.pal(9, "Blues")) )(255)


#### Plotting heatmap for Expression Visualization of all genes in each sample

#pdf(file="Output/gene_expression_heatmap_all.pdf")
pheatmap(sampleDistMatrix, #plot matrix of expression similarity
         clustering_distance_rows=sampleDists, #cluster rows
         clustering_distance_cols=sampleDists, #cluster columns
         col=colors) #set colors
#dev.off()

#### Plotting PCA from Expression visualization

#pdf(file="Output/PCA_all.pdf")
plotPCA(rld, intgroup = c("group")) #plot PCA of samples with all data
#dev.off()


```



# Differential Gene Expression Analysis
```{r}
#DESeq2 Test: test of the factor 

#run differential expression test by group using the wald test
DEG.int <- DESeq(data)

#save DE results
DEG.int.res <- results(DEG.int)

# Volcano Plot (significance as a function of fold change)
par(mar=c(5,5,5,5), cex=1.0, cex.main=1.4, cex.axis=1.4, cex.lab=1.4)
topT <- as.data.frame(DEG.int.res)
#Adjusted P values (FDR Q values)
with(topT, plot(log2FoldChange, padj, pch=20, main="Volcano plot", cex=1.0, xlab=bquote(~Log[2]~fold~change), ylab=bquote(padj)))
abline(h=0.05, col="black", lty=3, lwd=1.0)
#Color the significant points with log2 fold change >2 red ()
with(subset(topT, padj<0.05), points(log2FoldChange, padj, pch=20, col="red", cex=0.5))


  EnhancedVolcano(DEG.int.res,
    lab = rownames(DEG.int.res),
    x = 'log2FoldChange',
    y = 'pvalue')

#view DE results
resultsNames(DEG.int)

# save log fold change information
log2_changes <- DEG.int.res[,c(2,6)]

#identify the number of significant p values with 5%FDR (padj<0.05)
sig.num <- sum(DEG.int.res$padj <0.05, na.rm=T)

#identify signficant pvalues with 5%FDR
sig <- subset(DEG.int.res, padj<0.05,)

#subset list of sig transcripts from original count data
sig.list <- data[which(rownames(data) %in% rownames(sig)),]

#save DEG list
write.csv(counts(sig.list), file="Output/DEG_05FDR.all.csv")

#save DEG df
res_df <- as.data.frame(DEG.int.res)
res_df <- res_df %>% 
  subset(padj<0.05)


#subset upregulated at 31°C
up.31.sig <- res_df %>% 
  subset(padj<0.05) %>%
  subset(log2FoldChange>0) #negative values are higher in the Forereef

#Sanity check on expression direction. Backreef should be on average higher than Forereef
plotCounts(DEG.int, gene=rownames(up.31.sig)[1], intgroup="group") 

#subset list of sig transcripts from original count data
up.31.sig.list <- data[which(rownames(data) %in% rownames(up.31.sig)),]
up.31.sig.list <- as.data.frame(counts(up.31.sig.list))

#add annotation data
up.31.sig.list$Gene <- rownames(up.31.sig.list)
up.31.sig.list.annot <- left_join(up.31.sig.list, Annot,  by="Gene")



#subset upregulated at 28°C
up.28.sig <- res_df %>% 
  subset(padj<0.05) %>%
  subset(log2FoldChange<0) #negative values are higher in the Forereef

#Sanity check on expression direction. Backreef should be on average higher than Forereef
plotCounts(DEG.int, gene=rownames(up.28.sig)[1], intgroup="group") 

#subset list of sig transcripts from original count data
up.28.sig.list <- data[which(rownames(data) %in% rownames(up.28.sig)),]
up.28.sig.list <- as.data.frame(counts(up.28.sig.list))

#add annotation data
up.28.sig.list$Gene <- rownames(up.28.sig.list)
up.28.sig.list.annot <- left_join(up.28.sig.list, Annot,  by="Gene")


```

#Plotting Differentially Expressed Genes (DEGs) PCA
```{r}
#### Plotting DEG PCA

# set site colors
cols <- c("Larvae_31" = "red","Larvae_28" = "lightblue")

rsig <- rlog(sig.list, blind=FALSE)

PCA.plot.auto <- plotPCA(rsig, intgroup = c("group")) #Plot PCA of all samples for DEG only
PCA.plot.auto #view plot

PC.info <-PCA.plot.auto$data #extract plotting data

dev.off()
#plot PCA results
pdf(file="figures/Pverr_larvae_treatment_PCA.DEG.pdf")
PCA.plot <- PC.info %>%
ggplot(aes(x=PC1, y=PC2,color=group), group=group)+
  geom_point(size=3)+
  scale_color_manual(values=cols)+
  #ylim(-15,15)+
  #xlim(-35,35)+
  xlab(paste0(PCA.plot.auto$labels$x[1]))+
  ylab(paste0(PCA.plot.auto$labels$y[1]))+
  theme_bw()+
  theme(legend.position = c(0.8, 0.8),
         text = element_text(size=15))
PCA.plot 
dev.off()

```
#### Plotting DEG heatmap
```{r}
#### Plotting DEG heatmap

ann_colors = list(
  group = c("Larvae_31" = "red","Larvae_28" = "lightblue"))

#make an expression object
mat <- assay(rsig)#

#make dataframe
df <- data.frame(colData(rsig)[c("group")])

#pdf(file="Output/DEG_Heatmap.pdf")
pheatmap(mat, color=inferno(10), annotation_col = df, clustering_method = "average", scale="row",
                clustering_distance_rows="euclidean", show_rownames =F, cluster_cols=T,
                show_colnames =F, gaps_col = c(5)) #plot heatmap of all DEG by group
#dev.off()

#plot heatmap by site
pdf("output/Pverr_larvae_Temperature_DEG_heaptmap_relative.pdf")
Hmap <- pheatmap(mat, color=inferno(10), annotation_col = df, annotation_colors = ann_colors, clustering_method = "average", scale="row",  clustering_distance_rows="euclidean", show_rownames =F, fontsize_row=8, cluster_cols=T, show_colnames =F, gaps_col = c(5)) #plot heatmap of all DEG by group
Hmap 
dev.off()

```
#Plotting PCA and Heatmap together
```{r}
DEplots <- ggarrange(PCA.plot, as.grob(Hmap), ncol=2, labels = c("A)","B)"))
ggsave("figures/Pverr_larvae_DEplots_Temperature.pdf", DEplots, width = 16, height=8)
ggsave("figures/Pverr_larvae_DEplots_Temperature.png", DEplots, width = 16, height=8)


```


#Adding Annotation to DEGs
```{r}
DEGs <- res_df
DEGs$Gene <- rownames(DEGs)
Output <- left_join(DEGs,Annot,  by="Gene")
write.csv(Output,"output/Annotated_DEGS.csv")

Output <- Output %>% separate(Hit_description, into = c("Hit_description1", "Hit_description2"),
          sep = "\\(|\\)")

Output$label <- paste0(Output$Hit_description1, " (", Output$Gene, ")")
Output$label <- gsub("Pver_gene_", "", Output$label)

Output$label <- factor(Output$label, levels = Output$label[order(Output$log2FoldChange)])


Output %>% arrange(log2FoldChange) %>%
  ggplot(aes(x=label, y=log2FoldChange, colour=padj)) +
  ylim(-8,10)+
          geom_point(size=2)+
          scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=label, xend=label, y=0, yend=log2FoldChange)) +
   geom_vline(xintercept = 12 + 0.5, linetype = "dashed") + # Add vertical line
  coord_flip()+
  theme_classic()





```

# GO ENRICHMENT
# Gene Length
```{r}
calc.lengths <- read.table("data/Tagseq_zymo_genohub/Pver_genome_assembly_v1.0_fixed.gff3", header=F)
calc.lengths <- calc.lengths %>% filter(V3=="mRNA")
calc.lengths$length <- calc.lengths$V5 - calc.lengths$V4
calc.lengths$V9 <- str_after_nth(calc.lengths$V9, "\\;", 1)
calc.lengths$V9 <- gsub("geneID=", "", calc.lengths$V9)
calc.lengths <- calc.lengths[,c(9:10)]
colnames(calc.lengths)[1] <- "Gene"
```

#GO enrichment of DEGs that are HIGHER IN EXPRESSION at 31
```{r}

ALL<-row.names(data) #set the all transcripts list
DEG <- as.character(row.names(up.31.sig.list)) #set the enrichment test list
IDs <- row.names(data) #set ID names
LENGTH <-calc.lengths

#load GO terms for all genes in the genome
GO.terms <-Annot[,c(2,13)]
splitted <- strsplit(as.character(GO.terms$Gene_ontology_IDs), ";") #split into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.terms$Gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all GOs with each assigned transcript
GO.terms$v2 <- gsub(" GO:", "GO:",GO.terms$v2)

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DEG.vector <-c(t(DEG))
ID.vector <- LENGTH$Gene
LENGTH.vector <-LENGTH$length

gene.vector <-  as.integer(ID.vector%in%DEG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ID.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")


MF.INT <- subset(enriched.GO.05, ontology=="MF")
L31.MF.INT <- MF.INT[order(MF.INT$numDEInCat),]
CC.INT <- subset(enriched.GO.05, ontology=="CC")
L31.CC.INT <- CC.INT[order(CC.INT$numDEInCat),]
BP.INT <- subset(enriched.GO.05, ontology=="BP")
L31.BP.INT <- BP.INT[order(BP.INT$numDEInCat),]
#write.csv(ALL , file = "Output/Transcriptome_GOTerms.csv")
#write.csv(MF.INT , file = "Output/BR.MF_Sig_Enriched_GO.05_INT.csv")
#write.csv(CC.INT , file = "Output/BR.CC_Sig_Enriched_GO.05_INT.csv")
#write.csv(BP.INT , file = "Output/BR.BP_Sig_Enriched_GO.05_INT.csv")

```

#Plotting BP enriched in DEGS from Larvae at 31
```{r}
# similarity of GO terms in the DEG list for BP
simMatrix <- calculateSimMatrix(L31.BP.INT$category,
                                orgdb="org.Hs.eg.db", #human database
                                ont=c("BP"),
                                method="Rel")


scores <- setNames(-log(L31.BP.INT$over_represented_pvalue), L31.BP.INT$category)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.9,
                                orgdb="org.Hs.eg.db")
colnames(reducedTerms)[1] <- "category"

#conduct clustering of the similarity matrix
hc <- hclust(dist(simMatrix))

# identify the information for each tip label in the data
d1 <- data.frame(id=hc$labels,
                 node=hc$order) #,val=rnorm(30, sd=3))
colnames(d1)[1] <- "category"
colnames(reducedTerms)[1] <- "category"
d1 <- left_join(d1,reducedTerms, by="category")
d1 <- left_join(d1,L31.BP.INT, by="category")

# Reorder levels of category based on parentTerm
d1 <- d1 %>% 
  arrange(parentTerm) %>%
  mutate(category = factor(category, levels = unique(category)))

pdf("figures/L31_GO_Enrichment_semsim.pdf", width = 8, height = 10)
d1 %>% 
  ggplot(aes(x=category, y=over_represented_pvalue, colour=parentTerm)) +
  #ylim(-16,16)+
  geom_segment(aes(x=category, xend=category, y=0, yend=over_represented_pvalue, colour = parentTerm)) +
  geom_point(aes(size=numDEInCat))+
  scale_colour_discrete() + 
  coord_flip()+
  theme_classic() +
  theme(legend.position = "right",
        axis.title.y = element_blank(),
        axis.title.x = element_text(size = 14, face = "bold"),
        axis.text.x = element_text(size = 10))
dev.off()

```

#GO enrichment of DEGs that are HIGHER IN EXPRESSION at 28
```{r}

ALL<-row.names(data) #set the all transcripts list
DEG <- as.character(row.names(up.28.sig.list)) #set the enrichment test list
IDs <- row.names(data) #set ID names
LENGTH <-calc.lengths

#load GO terms for all genes in the genome
GO.terms <-Annot[,c(2,13)]
splitted <- strsplit(as.character(GO.terms$Gene_ontology_IDs), ";") #split into multiple GO ids
GO.terms <- data.frame(v1 = rep.int(GO.terms$Gene, sapply(splitted, length)), v2 = unlist(splitted)) #list all GOs with each assigned transcript
GO.terms$v2 <- gsub(" GO:", "GO:",GO.terms$v2)

#change contig lists to vectors
ALL.vector <-c(t(ALL))
DEG.vector <-c(t(DEG))
ID.vector <- LENGTH$Gene
LENGTH.vector <-LENGTH$length

gene.vector <-  as.integer(ID.vector%in%DEG.vector) #Construct new vector with 1 for DEG and 0 for others
names(gene.vector)=ID.vector #set names
DEG.pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) #weight vector by length of gene

#Find enriched GO terms, "selection-unbiased testing for category enrichment amongst differentially expressed (DE) genes for RNA-seq data"
GO.wall<-goseq(DEG.pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:CC", "GO:BP", "GO:MF"), method="Wallenius", use_genes_without_cat=TRUE)

#How many enriched GO terms do we have
class(GO.wall)
head(GO.wall)
tail(GO.wall)
nrow(GO.wall)

#Find only enriched GO terms that are statistically significant at cutoff 
enriched.GO.05.a<-GO.wall$category[GO.wall$over_represented_pvalue<.05]
enriched.GO.05<-data.frame(enriched.GO.05.a)
colnames(enriched.GO.05) <- c("category")
enriched.GO.05 <- merge(enriched.GO.05, GO.wall, by="category")


MF.INT <- subset(enriched.GO.05, ontology=="MF")
L28.MF.INT <- MF.INT[order(MF.INT$numDEInCat),]
CC.INT <- subset(enriched.GO.05, ontology=="CC")
L28.CC.INT <- CC.INT[order(CC.INT$numDEInCat),]
BP.INT <- subset(enriched.GO.05, ontology=="BP")
L28.BP.INT <- BP.INT[order(BP.INT$numDEInCat),]
#write.csv(ALL , file = "Output/Transcriptome_GOTerms.csv")
#write.csv(MF.INT , file = "Output/BR.MF_Sig_Enriched_GO.05_INT.csv")
#write.csv(CC.INT , file = "Output/BR.CC_Sig_Enriched_GO.05_INT.csv")
#write.csv(BP.INT , file = "Output/BR.BP_Sig_Enriched_GO.05_INT.csv")

```

# plotting DEGs 
```{r}

data2.labels <- data.frame(
  x = c(14, 11),
  y = c(6, -6),
  padj=c(0,0),
  text.lab = c("Higher at 31°C", "Higher at 28°C")
  )

pdf("figures/DEG_LFC_Larvae_Temperature.pdf", height = 12, width = 8)
Output %>% arrange(log2FoldChange) %>%
  ggplot(aes(x=label, y=log2FoldChange, colour=padj)) +
  ylim(-10,10)+
  geom_point(size=2)+
  scale_colour_gradientn(colours = colorspace::heat_hcl(7))+
  geom_segment(aes(x=label, xend=label, y=0, yend=log2FoldChange)) +
  geom_vline(xintercept = 12 + 0.5, linetype = "dashed") + # Add vertical line
  geom_text(data = data2.labels, aes(x = x, y = y, label = text.lab), color="black",size=5)+
  #geom_text(data = data.GOBP.labels, aes(x = x, y = y, label = text.lab), color="black",size=2.8)+
  #geom_text(data = data.GOMF.labels, aes(x = x, y = y, label = text.lab), color="blue",size=2.8)+
  coord_flip()+
  theme_classic() +
  theme(legend.position = c(0.8, 0.45),
        axis.title.y = element_blank(),
        axis.title.x = element_text(size = 14, face = "bold"))
dev.off()

```

# GO Plotting
#Plotting BP enriched in DEGS from Larvae at 28
```{r}

# similarity of GO terms in the DEG list for BP
simMatrix <- calculateSimMatrix(L28.BP.INT$category,
                                orgdb="org.Hs.eg.db", #human database
                                ont=c("BP"),
                                method="Rel")


scores <- setNames(-log(L28.BP.INT$over_represented_pvalue), L28.BP.INT$category)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.9,
                                orgdb="org.Hs.eg.db")
colnames(reducedTerms)[1] <- "category"

#conduct clustering of the similarity matrix
hc <- hclust(dist(simMatrix))

# identify the information for each tip label in the data
d1 <- data.frame(id=hc$labels,
                 node=hc$order) #,val=rnorm(30, sd=3))
colnames(d1)[1] <- "category"
colnames(reducedTerms)[1] <- "category"
d1 <- left_join(d1,reducedTerms, by="category")
d1 <- left_join(d1,L28.BP.INT, by="category")

# Reorder levels of category based on parentTerm
d1 <- d1 %>% 
  arrange(parentTerm) %>%
  mutate(category = factor(category, levels = unique(category)))

pdf("figures/L28_GO_Enrichment_semsim.pdf", width = 6, height = 6)
d1 %>% 
  ggplot(aes(x=category, y=over_represented_pvalue, colour=parentTerm)) +
  #ylim(-16,16)+
  geom_segment(aes(x=category, xend=category, y=0, yend=over_represented_pvalue, colour = parentTerm)) +
  geom_point(aes(size=numDEInCat))+
  scale_colour_discrete() + 
  coord_flip()+
  theme_classic() +
  theme(legend.position = "right",
        axis.title.y = element_blank(),
        axis.title.x = element_text(size = 14, face = "bold"),
        axis.text.x = element_text(size = 10))
dev.off()

```

```{r}








p <- ggtree(hc)

# Make the initial plot
pdf("figures/L31.BP.GOTree.pdf", height=6, width=12)
p %<+% d1  + geom_text(aes(label=node), hjust=-.3, size=2)+
  geom_hilight(node=98, fill="steelblue",  extendto = 0.3)+
  geom_cladelabel(node=98, label="  lipoate biosynthetic process", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  xlim(-4,1.8)+
  geom_hilight(node=103, fill="yellow",  extendto = 0.3)+
  geom_cladelabel(node=103, label="  chromosome organization involved in meiotic cell cycle", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=102, fill="darkgreen",  extendto = 0.3)+
  geom_cladelabel(node=102, label="  maintenance of epithelial cell apical/basal polarity", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=94, fill="green",  extendto = 0.3)+
  geom_cladelabel(node=94, label="  regulation of neurotransmitter secretion", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=95, fill="pink",  extendto = 0.3)+
    geom_cladelabel(node=95, label="  bleb assembly", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=96, fill="blue",  extendto = 0.3)+
    geom_cladelabel(node=96, label="  negative regulation of endocytosis", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=101, fill="cyan",  extendto = 0.3)+
    geom_cladelabel(node=101, label="  negative regulation of cell volume", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=100, fill="coral",  extendto = 0.3)+
    geom_cladelabel(node=100, label="Parent", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=104, fill="lightgreen",  extendto = 0.3)+
    geom_cladelabel(node=104, label="Parent", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=105, fill="orange",  extendto = 0.3)+
    geom_cladelabel(node=105, label="  neuron projection development
", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_hilight(node=92, fill="red",  extendto = 0.3)+
  geom_cladelabel(node=92, label="regulation of neurotransmitter secretion", 
                  color="black", offset=.3, align=TRUE, fontsize = 2)+
  geom_tiplab(aes(label=label), align=TRUE, size=4,linesize=0.1, as_ylab=TRUE, offset=.2)
  #geom_hilight(data = d1, mapping=aes(node = node, fill=parentTerm,  extendto = 0.5))
dev.off()

```

```{r}
# Make a second plot with the original, naming the new plot "dot", 
# using the data you just created, with a point geom.
p2 <- facet_plot(p, panel="Number DE in Term", data=d1, geom=geom_point, aes(x=numDEInCat, color=over_represented_pvalue))

# Make some more data with another random value.
d2 <- data.frame(id=tree$tip.label, value = abs(rnorm(30, mean=100, sd=50)))

# Now add to that second plot, this time using the new d2 data above, 
# This time showing a bar segment, size 3, colored blue.
p3 <- facet_plot(p2, panel='bar', data=d1, geom=geom_segment, 
           aes(x=0, xend=value, y=y, yend=y), size=3, color='blue4') 

# Show all three plots with a scale
p3 + theme_tree2()

```


